connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginsTest.java

```diff
@@ -185,6 +189,68 @@ public void shouldInstantiateAndConfigureDefaultHeaderConverter() {
         assertTrue(headerConverter instanceof SimpleHeaderConverter);
     }
 
+    @Test(expected = ConnectException.class)
+    public void shouldThrowIfPluginThrows() {
+        TestPlugins.assertInitialized();
+
+        assertNotNull(plugins.newPlugin(
+            TestPlugins.ALWAYS_THROW_EXCEPTION,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        ));
+    }
+
+    @Test(expected = ConnectException.class)
+    public void shouldCacheExceptionalClassInitialization() {
+        TestPlugins.assertInitialized();
+        // Attempt to load the class with the wrong classloader first
+        try {
+            ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+            assertFalse(classLoader instanceof PluginClassLoader);
+            Class<?> clazz = plugins.delegatingLoader().loadClass(TestPlugins.EXPECT_PLUGIN_CLASS_LOADER);
+            Plugins.newPlugin(classLoader, clazz);
+            fail("Should have thrown exception with wrong classloader");
+        } catch (ConnectException | ClassNotFoundException e) { }
```

Comment:
Nit: would help readability a bit to have a comment in the body of the `catch` block detailing that the exception is expected, something akin to:
```suggestion
        } catch (ConnectException | ClassNotFoundException e) { 
            // Should be thrown during a successful test
        }
```
Additionally, if there's an expected exception, we should ideally only expect one; catching both `ConnectException` and `ClassNotFoundException` is a little strange. Any reason both should be permitted during this part of the test instead of just catching the one that's currently expected to be thrown during a successful run?